proposals/schism-eval-fib/README.md

```diff
@@ -0,0 +1,204 @@
+   time they appear.  We could take a version of `eval` compiled from a
+   newer version of Schism that doesn't have this problem.
+
+ * We could use GC types to avoid calling out so much to the host.
+   Would need implementation of GC in an engine though.
+
+## Benchmark kind
+
+I think we can consider this benchmark to be a "kernel".
+
+## License
+
+```
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may
+not use this file except in compliance with the License.  You may obtain
+a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+```
+
+## How to build
+
+Source: https://github.com/google/schism/blob/master/test/eval.ss
+
+```
+mkdir /tmp/schism-eval-fib
+git clone https://github.com/google/schism
+cd schism
+# Anyref support pending merge: https://github.com/google/schism/pull/72
+git checkout 1b202ad59b7db45c95107f38ba5513cdaac421f7
+# FIXME: Build from self-hosted wasm instead of Guile.
+bootstrap-from-guile.sh /tmp/schism-eval-fib/schism-eval.wasm test/eval.ss
+cp rt/rt.mjs /tmp/schism-eval-fib/
+```
+
+Place this file in `/tmp/schism-eval-fib/run.mjs`:
+
+```js
+import fs from 'fs';
+import path from 'path';
+import process from 'process';
+
+import { Engine } from './rt.mjs';
+
+async function callAndThenExit(f) {
+    // Pending promises don't keep Node alive; you need timeouts for
+    // that.  A bit hacky, but whatever.
+    const timer = setTimeout(() => {}, 999999);
+    try {
+        await f();
+    } catch (e) {
+        console.warn('Test failed: ' + e);
+        console.warn(e.stack);
+        process.exit(1);
+    }
+    process.exit(0);
+}
+
+async function run() {
+    const engine = new Engine;
+    const file = path.dirname(process.argv[1]) + '/schism-eval.wasm';
+    const bytes = fs.readFileSync(file);
+    const wasm = await engine.loadWasmModule(bytes);
+    engine.setCurrentInputPortChars(`
+      (begin
+        (display
+         (let fib ((n 25))
+           (if (< n 2)
+               n
+               (+ (fib (- n 1)) (fib (- n 2))))))
+        (newline))`);
+
+    wasm.exports['read-and-eval']();
+}
+
+callAndThenExit(run);
+```
+
+Note that this `run.mjs` test runner uses Node APIs.  We should
+certainly factor out all of this to some helper library.
+
+## How to run
+
+To run in Node/v8, use:
+
+```
+node --experimental-modules --experimental-wasm-anyref \
+    ./run.mjs
+```
+
+## Expected output
+
+The benchmark suite should verify that the node invocation prints the
+following:
+
+```
+75025
+```
+
+## Expected latency
+
+On my system this test runs in about 4 or 5 seconds:
+
+```
+real	0m4.729s
```

Comment:
Cool. I assume since this is fib that configuring different inputs for different benchmark sizes.

What do you think of adding recommending small, medium and large inputs?

Reply:
Right, no problem; will do tomorrow.

Do you want benchmarks in a machine-readable format somehow?

Would you like a benchmark runner script, or do you have one in mind?